src/components/logic/calculate.js

```diff
@@ -0,0 +1,34 @@
+import operate from './operate';
+
+const type = (e) => {
+  const form = document.forms[0];
+  form[0].value += e.target.innerText;
+};
+
+const clear = () => {
+  const form = document.forms[0];
+  form[0].value = '';
+};
+
+const calculate = () => {
+  const form = document.forms[0];
+  const equation = form[0].value.match(/\d+|\D+/g);
+  let index = 0;
+  while (index !== equation.length) {
+    if (equation[index] === '-') {
+      equation[index + 1] *= (-1);
+    }
+
+    if (!Number.isNaN(parseInt(equation[index], 10))) {
+      equation[index] *= (-1) * (-1);
+    }
+
+    if (typeof equation[index] === 'string' && index > 0) {
+      equation[index + 1] = operate(equation[index - 1], equation[index + 1], equation[index] === '-' ? '+' : equation[index]);
+      form[0].value = equation[index + 1];
+    }
+    index += 1;
+  }
+};
+
+export { calculate, type, clear };
```

Comment:
- [ ] I noticed that your calculate.js file is different from the one provided to you here [calculate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js)  to use in your project as it mentioned on the requirement. Please fix it :+1: 
![Screenshot from 2022-09-06 22-24-31](https://user-images.githubusercontent.com/6615394/188751920-7b41cb8e-d591-4afd-8fa9-c084f95a9ef3.png)